Automattic/wp-calypso
# Introduce component usage stats

## What?
This PR introduces [`react-scanner`](https://www.npmjs.com/package/react-scanner) to keep an eye on component usage stats.

## Why?
It can be useful to track the adoption of our components over time, and we're automating that as part of our work on design systems (bjpUB-p2).

See https://github.com/WordPress/gutenberg/pull/65463 for the same in Gutenberg itelf.

## How?

Running `react-scanner` with the default config to count components and props yields a result like this:

```json
{
 "LinkPicker": {
    "instances": [
      {
        "importInfo": {
          "imported": "LinkPicker",
          "local": "LinkPicker",
          "moduleName": "@wordpress/components",
          "importType": "ImportSpecifier"
        },
        "props": {
          "value": "(Identifier)",
          "onLinkPicked": "(Identifier)",
          "onCancel": "(Identifier)"
        },
        "propsSpread": false,
        "location": {
          "file": "gutenberg/packages/format-library/src/link/modal-screens/link-picker-screen.native.js",
          "start": {
            "line": 52,
            "column": 4
          }
        }
      }
    ]
  },
  "CardHeader": {
    "instances": [
      {
        "importInfo": {
          "imported": "CardHeader",
          "local": "CardHeader",
          "moduleName": "@wordpress/components",
          "importType": "ImportSpecifier"
        },
        "props": {
          "isBorderless": false,
          "justify": "left",
          "size": "small",
          "gap": "6"
        },
        "propsSpread": false,
        "location": {
          "file": "gutenberg/packages/preferences/src/components/preferences-modal-tabs/index.js",
          "start": {
            "line": 142,
            "column": 10
          }
        }
      }
    ]
  }
}
```

## Testing Instructions
* Checkout this branch
* Run `yarn install`
* Run `yarn run component-usage-stats`
* Check out the output

### Testing Instructions for Keyboard
Same

## Screenshots or screencast 
None
